test/js/bun/s3/s3.test.ts

```diff
@@ -0,0 +1,443 @@
+import { describe, expect, it, beforeAll, afterAll } from "bun:test";
+import { getSecret } from "harness";
+import { randomUUID } from "crypto";
+import { S3, s3, file } from "bun";
+import type { S3File, S3FileOptions } from "bun";
+
+const s3Options: S3FileOptions = {
+  accessKeyId: getSecret("S3_R2_ACCESS_KEY"),
+  secretAccessKey: getSecret("S3_R2_SECRET_KEY"),
+  endpoint: getSecret("S3_R2_ENDPOINT"),
+};
+
+const S3Bucket = getSecret("S3_R2_BUCKET");
+
+// 15 MiB big enough to Multipart upload in more than one part
+const bigPayload = Buffer.alloc(15 * 1024 * 1024, "a");
```

Comment:
this should be a longer repeating pattern so that we can verify the data is correct. maybe the contents of this file itself would be a good way to do that.